### DISCUSSION TITLE
How to modify the buffer height when debugging the console exe??

### DISCUSSION THREAD

**Discussion by wxuefei** [microsoft/terminal#15818] [Q&A] [answered]
# Description of the new feature/enhancement

I can modify the console buffer height use the below code when I debugging the console exe with VC  & cmd, but it doesn't work in WindowsTerminal,  the buffer height seems always is 9030,  how to  modify it ?

    system("mode con:cols=120 lines=20000");
or

    SetConsoleScreenBufferSize(GetStdHandle(STD_OUTPUT_HANDLE), { 120,20000 });

# Proposed technical implementation details (optional)

**Comment by zadjii-msft** [comment, marked as answer]
This isn't something that's possible with the Terminal, or really something we're planning on supporting. Out of curiosity, why would you like to change the history size at runtime from a CLI application/?

(changing the windows dimensions is tracked in #5094, but that's definitely not the buffer height)

**Reply by wxuefei** [reply]
When you debugging an exe  that outputs a lot of logs, you will need it !  Why cmd works well ,but windowsterminal doesn't?  is it very diffcult?

**Reply by zadjii-msft** [reply]
Yea, but this doesn't _really_ seem like something that needs to be changeable _at runtime_, _from a cli application_. I'd just slap a `"historySize": 99999` in my [`settings.json`](https://learn.microsoft.com/en-us/windows/terminal/customize-settings/profile-advanced#history-size), and forget about it. That'll just max out the history size, and you'll never need to think about resizing it again.

Am I missing something?

**Reply by wxuefei** [reply]
OK！ I got it ! There are two  “historySize”, one in defaults section, it be used for debugging terminal window,  another historySize in list{cmd}  section,  it be used for terminal shell. Thanks !
